RandBLAS/sparse.hh

```diff
@@ -140,14 +251,16 @@ static void sketch_cscrow(
     int64_t ldb,
     int threads
```

Comment:
This used to have a default value of 4. Can we add that back in? I'd like for ``lskges`` to have the same signature as ``lskge3``, and the latter doesn't have a required ``threads`` argument.

Reply:
Sure. I will add it back in. What  are your thoughts on making the default 1? 

For instance a default of N > 1 threads will not yield good performance if there are M < N physical cores present. On the other hand if you go with OpenMP's default then it allocates a thread for each logical core (hyperthread). this is usually bad for performance. the best performance for numerical codes is almost always 1 thread per physical core. When MPI is in play (thinking about the future) OpenMP thread pools should be sized such that the threads in each MPI process are uniquely mapped to physical cores across a node. With P MPI ranks per node, and M physical cores per node, a default of N > 1 OpenMP threads will result in poor performance when N * P > M.

Reply:
The usual way to specify number of threads is the `OMP_NUM_THREADS` and `OMP_PLACES` environment variables. `omp_set_num_threads` documentation overrides indicates that it overrides those controls, so I think we should avoid using it, or at the very least check in the environment for `OMP_NUM_THREADS` and only call it if it is not set.

Reply:
It's probably best if we never call `omp_set_num_threads` internally in the library and leave it up to the code calling the library to do it right. this is where documentation becomes critical.

Reply:
If you say it's best that we never call ``omp_set_num_threads`` in the library, that's fine. Can the user call ``omp_set_num_threads`` before and after a call to a RandBLAS function they want to benchmark? That is, can the user do something like the following?
```
omp_set_num_threads(4);
lskges(....);
omp_set_num_threads(... default ...);
```

Reply:
Seems the answer to my question is yes: https://stackoverflow.com/questions/22813923/difference-between-num-threads-vs-omp-set-num-threads-vs-omp-num-threads.